appstudio-controller/controllers/appstudio.redhat.com/environment_controller_test.go

```diff
@@ -340,10 +340,98 @@ var _ = Describe("Environment controller tests", func() {
+						},
+					},
+				},
+			}
+			err := k8sClient.Create(ctx, &env)
+			Expect(err).To(BeNil())
+
+			By("reconciling the Environment")
+			req := ctrl.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      env.Name,
+					Namespace: env.Namespace,
+				},
+			}
+			_, err = reconciler.Reconcile(ctx, req)
+			Expect(err).To(BeNil())
+
+			err = k8sClient.Get(ctx, client.ObjectKeyFromObject(&env), &env)
+			Expect(err).To(BeNil())
+			expectEnvironmentStatusConditionError("the secret secret-that-doesnt-exist referenced by the Environment resource was not found", EnvironmentReasonSecretNotFound, env)
+
+			By("fixing the error")
+			secret := corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-my-managed-env-secret",
+					Namespace: apiNamespace.Name,
+				},
+				Type: sharedutil.ManagedEnvironmentSecretType,
+				Data: map[string][]byte{
+					"kubeconfig": ([]byte)("{}"),
+				},
+			}
+			err = k8sClient.Create(ctx, &secret)
+			Expect(err).To(BeNil())
+
+			env.Spec.UnstableConfigurationFields.ClusterCredentialsSecret = secret.Name
+			Expect(k8sClient.Update(ctx, &env)).To(Succeed())
+
+			By("reconciling again")
+			_, err = reconciler.Reconcile(ctx, req)
+			Expect(err).To(BeNil())
+
+			err = k8sClient.Get(ctx, client.ObjectKeyFromObject(&env), &env)
+			Expect(err).To(BeNil())
+			Expect(env.Status.Conditions[0].Type).To(Equal(EnvironmentConditionErrorOccurred))
+			Expect(env.Status.Conditions[0].Status).To(Equal(metav1.ConditionFalse))
+			Expect(env.Status.Conditions[0].Reason).To(Equal(EnvironmentReasonErrorOccurred + "Resolved"))
+			Expect(env.Status.Conditions[0].Message).To(BeEmpty())
+			Expect(env.Status.Conditions[1].Type).To(Equal(EnvironmentConditionReconciled))
+			Expect(env.Status.Conditions[1].Status).To(Equal(metav1.ConditionTrue))
+			Expect(env.Status.Conditions[1].Reason).To(Equal(EnvironmentReasonSecretNotFound + "Resolved"))
+			Expect(env.Status.Conditions[1].Message).To(BeEmpty())
+
+			By("ensuring that another reconcile has no effect on the condition")
+			_, err = reconciler.Reconcile(ctx, req)
+			Expect(err).To(BeNil())
+
+			err = k8sClient.Get(ctx, client.ObjectKeyFromObject(&env), &env)
+			Expect(err).To(BeNil())
+			Expect(env.Status.Conditions[0].Type).To(Equal(EnvironmentConditionErrorOccurred))
+			Expect(env.Status.Conditions[0].Status).To(Equal(metav1.ConditionFalse))
+			Expect(env.Status.Conditions[0].Reason).To(Equal(EnvironmentReasonErrorOccurred + "Resolved"))
+			Expect(env.Status.Conditions[0].Message).To(BeEmpty())
+			Expect(env.Status.Conditions[1].Type).To(Equal(EnvironmentConditionReconciled))
+			Expect(env.Status.Conditions[1].Status).To(Equal(metav1.ConditionTrue))
+			Expect(env.Status.Conditions[1].Reason).To(Equal(EnvironmentReasonSecretNotFound + "Resolved"))
+			Expect(env.Status.Conditions[1].Message).To(BeEmpty())
+		})
+
 		It("should not return an error if the Environment does not container UnstableConfigurationFields", func() {
```

Comment:
typo : `container --> contains,` this typo is not from your changes, I think can be fixed in this PR

Reply:
Fixed.